### PR TITLE
Process special magic tags but allow "empty" values for numbers.

### PR DESCRIPTION
**Pull request by JoryHogeveen**
Fixes: #5697 

Not sure if it needs unit tests. @sc0ttkclark ?

## Description

The Pods documentation shows the use of special magic tags. However, this isn't working in Pods Templates. Not even when using pods shortcodes within pods templates.

As stated in the documentation, this should work:
https://docs.pods.io/displaying-pods/magic-tags/special-magic-tags/#Situations_Where_Special_Magic_Tags_Can_Be_Used

> In the ‘where’/ ‘having’ field of Pods shortcodes

This PR will fallback on these special magic tags if no valid field value has been found.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.
- [ ] My code includes PHP Unit Tests (if applicable)

### CODE REVIEW

File: classes/Pods.php [hunk 1]
```diff
@@ -4175,6 +4175,11 @@ private function process_magic_tags( $tag ) {
 			$value = $this->display( $field_name );
 		}
 
+		// Process special magic tags but allow "empty" values for numbers.
+		if ( ! $value && ! is_numeric( $value ) ) {
```
Comment (sc0ttkclark) [inline classes/Pods.php:4179]:
```suggestion
		if ( ! $value && ! is_numeric( $value ) && null === $this->fields( $field_name ) ) {
```

File: classes/Pods.php [hunk 2]
```diff
@@ -4175,6 +4175,11 @@ private function process_magic_tags( $tag ) {
 			$value = $this->display( $field_name );
 		}
 
+		// Process special magic tags but allow "empty" values for numbers.
+		if ( ! $value && ! is_numeric( $value ) && pods_shortcode_allow_evaluate_tags() && null === $this->fields( $field_name ) ) {
```
Comment (sc0ttkclark) [inline classes/Pods.php:4179]:
```suggestion
		if (
			! $value
			&& ! is_numeric( $value )
			&& pods_shortcode_allow_evaluate_tags()
			&& null === $this->fields( $field_name )
		) {
```

**Comment by JoryHogeveen** [comment]
@jimtrue Can you think of issues that might rise from this change? If not, let's merge :)

File: classes/Pods.php [hunk 3]
```diff
@@ -4175,6 +4175,16 @@ private function process_magic_tags( $tag ) {
...
 		}
 
+		// Process special magic tags but allow "empty" values for numbers.
+		if (
+			! $value
+			&& ! is_numeric( $value )
+			&& pods_shortcode_allow_evaluate_tags()
+			&& ! $this->fields( $field_name )
+		) {
+			$value = pods_evaluate_tag( $field_name );
```
Comment (JoryHogeveen) [inline classes/Pods.php:4185]: Sanitize output:
```suggestion
			$value = pods_evaluate_tag( $field_name, true );
```

**Comment by sc0ttkclark** [comment]
Let's hold on this for further review

**Comment by JoryHogeveen** [comment]
@sc0ttkclark Still I'd prefer we finish #5735 (that PR includes the commits from this one). It's a far more complete solution.

**Comment by sc0ttkclark** [comment]
Needs conflict resolution, you did both PRs so I'm going to leave that for you to confirm and then follow up with QA @JoryHogeveen

**Comment by JoryHogeveen** [comment]
All commits from this PR are already merged by #5735 
Closing PR.

**Pull request closed without merging**